src/js/coblocks-counter.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import counterUp from 'counterup2';
+
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+
+const ACTIVE = 'coblocks-counter-active';
+class CoBlocksCounter {
+	callback = ( entries ) => {
+		entries.forEach( ( entry ) => {
+			const el = entry.target;
+			const counterSpeed = parseInt( el?.getAttribute( 'data-counter-speed' ) || 1 ) * 1000;
+			if ( ! el.classList.contains( ACTIVE ) && entry.isIntersecting ) {
+				setTimeout( () => {
+					counterUp( el, { duration: counterSpeed } );
+					el.classList.add( ACTIVE );
+				}, 200 );
+			}
+		} );
+	}
+
+	init = () => {
+		const els = document.querySelectorAll( '.wp-block-coblocks-counter' );
+
+		// If reduce motion is on, don't animate.
+		const reduceMotion = window.matchMedia( '(prefers-reduced-motion: reduce)' ).matches;
```

Comment:
It's easy to forget about this kind of thing. Glad to see it here! We should do this for our carousel's as well.